lib/arjdbc/abstract/database_statements.rb

```diff
@@ -5,63 +5,25 @@ module Abstract
     # database for JDBC based adapters
     module DatabaseStatements
 
-      # Executes a delete statement in the context of this connection.
-      # @param sql the query string (or AREL object)
-      # @param name logging marker for the executed SQL statement log entry
-      # @param binds the bind parameters
-      # @override available since **AR-3.1**
-      def exec_delete(sql, name, binds)
-        if prepared_statements?
-          log(sql, name || 'SQL', binds) { @connection.execute_delete(sql, binds) }
+      # It appears that at this point (AR 5.0) "prepare" should only ever be true
+      # if prepared statements are enabled
+      def exec_query(sql, name = nil, binds = [], prepare: false)
+        if without_prepared_statement?(binds)
+          execute(sql, name)
         else
-          sql = to_sql(sql, binds) if sql.respond_to?(:to_sql)
-          log(sql, name || 'SQL') { @connection.execute_delete(sql) }
+          binds = convert_legacy_binds_to_attributes(binds) if binds.first.is_a?(Array)
```

Comment:
why we need to do this, is it only due some tests (maybe they're stupid tests that would not pass on MRI) ?

Reply:
Unfortunately, they have a test that explicitly tests that it can still handle legacy bind parameters, so I'm not sure why its there but seems to be intentional for now.

```
    if ActiveRecord::Base.connection.prepared_statements
      def test_select_all_with_legacy_binds
        post = Post.create!(title: "foo", body: "bar")
        expected = @connection.select_all("SELECT * FROM posts WHERE id = #{post.id}")
        result = @connection.select_all("SELECT * FROM posts WHERE id = #{Arel::Nodes::BindParam.new.to_sql}", nil, [[nil, post.id]])
        assert_equal expected.to_hash, result.to_hash
      end
    end
```